sources/packages/backend/libs/integrations/src/services/sin-validation/sin-validation.service.ts

```diff
@@ -74,120 +79,192 @@ export class SINValidationService extends RecordDataModelService<SINValidation>
    */
   async updateSINValidationFromESDCResponse(
     validationResponse: SINValidationFileResponse,
-    isValidSIN: boolean,
     receivedFileName: string,
     processDate: Date,
     auditUserId: number,
   ): Promise<SINValidationUpdateResult> {
-    let operationDescription: string;
-    // If the list of the selected columns must be changed please keep in mind that
-    // these fields are also the ones used later to "clone" the record if needed,
-    // as explained further along the method.
-    const existingValidation = await this.repo
-      .createQueryBuilder("sinValidation")
-      .select([
-        "sinValidation.id",
-        "sinValidation.sin",
-        "sinValidation.sinStatus",
-        "sinValidation.dateReceived",
-        "sinValidation.validSINCheck",
-        "sinValidation.dateSent",
-        "sinValidation.fileSent",
-        "sinValidation.givenNameSent",
-        "sinValidation.surnameSent",
-        "sinValidation.dobSent",
-        "sinValidation.genderSent",
-        "sinValidation.temporarySIN",
-        "sinValidation.sinExpiryDate",
-        "student.id",
-      ])
-      .innerJoin("sinValidation.student", "student")
-      .where("sinValidation.id = :sinValidationId", {
-        sinValidationId: validationResponse.referenceIndex,
-      })
-      .getOne();
+    const isValidSIN =
+      validationResponse.sinCheckStatus === SINCheckStatus.Passed;
+
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        const auditUser = await this.systemUsersService.systemUser();
+        let operationDescription: string;
+        // If the list of the selected columns must be changed please keep in mind that
+        // these fields are also the ones used later to "clone" the record if needed,
+        // as explained further along the method.
+        const existingValidation = await this.repo
```

Comment:
IMO it is always better to use the select from transactionalEntityManager `transactionalEntityManager.getRepository(Entity)` when a select is done inside the transaction.